src/bulk/common.ts

```diff
@@ -167,59 +163,48 @@ export class Batch<T = Document> {
   }
 }
 
+function generateIdMap(ids: Document[]): { [key: number]: any } {
+  const idMap: { [index: number]: any } = {};
+  for (const doc of ids ?? []) {
+    idMap[doc.index] = doc._id;
+  }
+  return idMap;
+}
 /**
  * @public
  * The result of a bulk write.
  */
 export class BulkWriteResult {
-  /** @deprecated Will be removed in 5.0 */
-  result: BulkResult;
-
-  /**
-   * Create a new BulkWriteResult instance
-   * @internal
-   */
-  constructor(bulkResult: BulkResult) {
-    this.result = bulkResult;
-  }
-
+  private result: BulkResult;
```

Comment:
All the bulk write result properties used to be getters and as a result they were readonly.  I think we should keep this behavior.

We could mark the properties as `readonly`.  This only impacts Typescript users but I think that's alright.

Reply:
Changed all properties to readonly.